securesubmit/services/gateway.py

```diff
@@ -145,7 +147,13 @@ def do_transaction(self, transaction, client_transaction_id=None):
 
             request_headers = {'Content-type': 'text/xml; charset=UTF-8',
                                'Content-length': str(len(xml))}
-            request = http.request('POST', self._url, headers=request_headers, body=xml)
+
+            request = None
+            if self._timeout is None:
```

Comment:
Set the timeout to a disabled value by default in the configuration. Then you can get rid of the check on timeout here and just always pass it.